script/testing/util/test_server.py

```diff
@@ -153,6 +159,16 @@ def run_test(self, test_case: TestCase):
         # run the pre test tasks
         test_case.run_pre_test()
 
+        # start a thread to collect the memory info if needed
+        if self.collect_mem_info:
+            # spawn a thread to collect memory info
+            self.collect_mem_thread = PeriodicTask(
+                self.collect_mem_freq, update_mem_info, self.db_process.pid,
+                self.collect_mem_freq, test_case.mem_metrics.mem_info_dict)
+            # collect the initial memory info
+            update_mem_info(self.db_process.pid, self.collect_mem_freq,
```

Comment:
Do we need this? It seems like it would collect info when the OLTPBench test isn't running

Reply:
Yes it is needed. Because the 1st execution of the `PeriodicTask` takes place after 5 sec, and we will miss the 0 sec data point if this line of code is removed.

Reply:
but wouldn't the 0 sec point be wrong here cause OLTPBench wont be running yet. I wonder what throughput is for 0 sec interval cause it should be 0 right. If its not zero it must be the first chance it gets to gather data after OLTPBench starts.

Reply:
What if the PeriodicTask `start` looked like:
```
    def start(self):
        """
        Schedule and start the initial Timer thread
        """
        
        if not self.is_running:
            self.function(*self.args, **self.kwargs)    #This handles the second 0

        self._timer = Timer(self.interval, self._run)

    def _run(self):
        self.is_running = True
        self.start()
        self.function(*self.args, **self.kwargs)
```

Reply:
Good approach! I will do it this way.

Reply:
Oh wait, there is a problem about the `is_running`. The `is_running` does not mean `is_not_started`, but means whether there is an existing Timer scheduled.